lib/iris/tests/unit/fileformats/netcdf/test_Saver.py

```diff
@@ -325,6 +330,154 @@ def test_valid_max_saved(self):
+
+    def test_mask_lazy_fill_value(self):
+        # Test that masked lazy data saves correctly when given a fill value.
+        index = (1, 1)
+        fill_value = 12345.
+        cube = self._make_cube('>f4', masked_index=index, lazy=True)
+        with self._netCDF_var(cube, fill_value=fill_value) as var:
+            self.assertEqual(var._FillValue, fill_value)
+            self.assertTrue(var[index].mask)
+
+    def test_mask_lazy_default_fill_value(self):
+        # Test that masked lazy data saves correctly when given a fill value.
+        index = (1, 1)
+        cube = self._make_cube('>f4', masked_index=index, lazy=True)
+        with self._netCDF_var(cube) as var:
+            self.assertNotIn('_FillValue', var.ncattrs())
+            self.assertTrue(var[index].mask)
+
+    def test_contains_fill_value_passed(self):
+        # Test that a warning is raised if the data contains the fill value.
+        cube = self._make_cube('>f4')
+        fill_value = 1
+        with self._warning_check(
+                'contains data points equal to the fill value'):
+            with self._netCDF_var(cube, fill_value=fill_value):
+                pass
+
+    def test_contains_fill_value_byte(self):
+        # Test that a warning is raised if the data contains the fill value
+        # when it is of a byte type.
+        cube = self._make_cube('>i1')
+        fill_value = 1
+        with self._warning_check(
+                'contains data points equal to the fill value'):
+            with self._netCDF_var(cube, fill_value=fill_value):
+                pass
+
+    def test_contains_default_fill_value(self):
+        # Test that a warning is raised if the data contains the default fill
+        # value if no fill_value argument is supplied.
+        cube = self._make_cube('>f4')
+        cube.data[0, 0] = nc.default_fillvals['f4']
+        with self._warning_check(
+                'contains data points equal to the fill value'):
+            with self._netCDF_var(cube):
+                pass
+
+    def test_contains_default_fill_value_byte(self):
+        # Test that no warning is raised if the data contains the default fill
+        # value if no fill_value argument is supplied when the data is of a
+        # byte type.
+        cube = self._make_cube('>i1')
+        with self._warning_check(
+                'contains data points equal to the fill value', False):
+            with self._netCDF_var(cube):
+                pass
+
+    def test_contains_masked_fill_value(self):
+        # Test that no warning is raised if the data contains the fill_value at
+        # a masked point.
+        fill_value = 1
+        cube = self._make_cube('>f4', masked_value=fill_value)
+        with self._warning_check(
+                'contains data points equal to the fill value', False):
+            with self._netCDF_var(cube, fill_value=fill_value):
+                pass
+
+    def test_masked_byte_default_fill_value(self):
```

Comment:
Possibly a missing testcase here ?
 = saving masked byte data _with a nonstandard fill-value_

It seems that the existing code this will _not_ generate a warning on save for this.
I'm not too clear what you would get when you read it back.
I'm not totally clear what it "ought" to do ?